src/System.Numerics.Vectors/tests/GenericVectorTests.netcoreapp.cs

```diff
@@ -0,0 +1,233 @@
+        [Fact]
+        public void ArrayBasedConstructorWithLessElements_Int64() => Assert.Throws<IndexOutOfRangeException>(() => TestArrayBasedConstructorWithLessElements<Int64>());
+        [Fact]
+        public void ArrayBasedConstructorWithLessElements_Single() => Assert.Throws<IndexOutOfRangeException>(() => TestArrayBasedConstructorWithLessElements<Single>());
+        [Fact]
+        public void ArrayBasedConstructorWithLessElements_Double() => Assert.Throws<IndexOutOfRangeException>(() => TestArrayBasedConstructorWithLessElements<Double>());
+
+        private void TestArrayBasedConstructorWithLessElements<T>() where T : struct
+        {
+            T[] values = GenerateRandomValuesForVector<T>(Vector<T>.Count - 1).ToArray();
+            Vector<T> v = new Vector<T>(values);
+        }
+
+        [Fact]
+        public void ArrayIndexBasedConstructorLessElements_Byte() => Assert.Throws<IndexOutOfRangeException>(() => TestArrayIndexBasedConstructorLessElements<Byte>());
+        [Fact]
+        public void ArrayIndexBasedConstructorLessElements_SByte() => Assert.Throws<IndexOutOfRangeException>(() => TestArrayIndexBasedConstructorLessElements<SByte>());
+        [Fact]
+        public void ArrayIndexBasedConstructorLessElements_UInt16() => Assert.Throws<IndexOutOfRangeException>(() => TestArrayIndexBasedConstructorLessElements<UInt16>());
+        [Fact]
+        public void ArrayIndexBasedConstructorLessElements_Int16() => Assert.Throws<IndexOutOfRangeException>(() => TestArrayIndexBasedConstructorLessElements<Int16>());
+        [Fact]
+        public void ArrayIndexBasedConstructorLessElements_UInt32() => Assert.Throws<IndexOutOfRangeException>(() => TestArrayIndexBasedConstructorLessElements<UInt32>());
+        [Fact]
+        public void ArrayIndexBasedConstructorLessElements_Int32() => Assert.Throws<IndexOutOfRangeException>(() => TestArrayIndexBasedConstructorLessElements<Int32>());
+        [Fact]
+        public void ArrayIndexBasedConstructorLessElements_UInt64() => Assert.Throws<IndexOutOfRangeException>(() => TestArrayIndexBasedConstructorLessElements<UInt64>());
+        [Fact]
+        public void ArrayIndexBasedConstructorLessElements_Int64() => Assert.Throws<IndexOutOfRangeException>(() => TestArrayIndexBasedConstructorLessElements<Int64>());
+        [Fact]
+        public void ArrayIndexBasedConstructorLessElements_Single() => Assert.Throws<IndexOutOfRangeException>(() => TestArrayIndexBasedConstructorLessElements<Single>());
+        [Fact]
+        public void ArrayIndexBasedConstructorLessElements_Double() => Assert.Throws<IndexOutOfRangeException>(() => TestArrayIndexBasedConstructorLessElements<Double>());
+
+        private void TestArrayIndexBasedConstructorLessElements<T>() where T : struct
+        {
+            T[] values = GenerateRandomValuesForVector<T>(Vector<T>.Count * 2).ToArray();
+            Vector<T> v = new Vector<T>(values, Vector<T>.Count + 1);
```

Comment:
I think we could do some verification here to ensure the Vector has the right values in it.

Reply:
Oh, I see it now.

I think the `Assert.Throws` should be moved inside this method and only around the `new Vector<T>` call.  Both to reduce duplication and to ensure the exact call is throwing the exception, and not the `GenerateRandomValuesForVector` method.

---
In reply to: [173288332](https://github.com/dotnet/corefx/pull/26499#discussion_r173288332) [](ancestors = 173288332)

Reply:
Done! Here and below, I actually wrote the `Span<T>` based constructor tests first and then made a copy to make them for `T[]` based constructor, when I didn't think much :-)